src/main/java/com/gradle/ccud/adapters/BuildScanObfuscationAdapter.java

```diff
@@ -0,0 +1,17 @@
+package com.gradle.ccud.adapters;
+
+import java.net.InetAddress;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Adapter for {@link com.gradle.develocity.agent.gradle.scan.BuildScanDataObfuscationConfiguration} and {@link com.gradle.ccud.adapters.enterprise.proxies.BuildScanDataObfuscationProxy}
+ */
+public interface BuildScanObfuscationAdapter {
```

Comment:
💬 Some adapter interfaces are the same as proxy interfaces. We can use just one, but I found the duplication helpful in this case. I got lost a couple of times in the proxy logic when the interface is used both as the adapter and the proxy. This also can be more easily enforced with architecture tests, which will be added in a later PR. Let me know if you feel strongly about de-duplicating interfaces, and I can revisit this.